platform/ext/target/nordic_nrf/common/native_drivers/spu.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "spu.h"
+#include <hal/nrf_spu.h>
+#include "region_defs.h"
+
+/* Platform-specific configuration */
+#define FLASH_SECURE_ATTRIBUTION_REGION_SIZE SPU_FLASH_REGION_SIZE
+#define SRAM_SECURE_ATTRIBUTION_REGION_SIZE  SPU_SRAM_REGION_SIZE
+
+#define NUM_FLASH_SECURE_ATTRIBUTION_REGIONS \
+	(FLASH_TOTAL_SIZE / FLASH_SECURE_ATTRIBUTION_REGION_SIZE)
+#define NUM_SRAM_SECURE_ATTRIBUTION_REGIONS \
+	(TOTAL_RAM_SIZE / SRAM_SECURE_ATTRIBUTION_REGION_SIZE)
+
+/* Convenience macros for SPU NCS attribution */
+#define FLASH_NSC_REGION_FROM_ADDR(addr) \
+	((u32_t)addr / FLASH_SECURE_ATTRIBUTION_REGION_SIZE)
+
+#define FLASH_NSC_SIZE_FROM_ADDR(addr) (FLASH_SECURE_ATTRIBUTION_REGION_SIZE \
+	- (((u32_t)(addr)) % FLASH_SECURE_ATTRIBUTION_REGION_SIZE))
+
+#define FLASH_NSC_SIZE_REG(size) ((31 - __builtin_clz(size)) - 4)
```

Comment:
Some comment explaining briefly this implementation would be nice here.

Reply:
Comments added now.
The description was minimal, following what we had in NCS, but agree, there's always room for improvement :)